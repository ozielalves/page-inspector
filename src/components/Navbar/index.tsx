import React from "react";
import styled from "styled-components";
import { ReactComponent as OzielLogo } from "../../assets/logo-navbar-oziel.svg";
import { ReactComponent as LinkedinLogo } from "../../assets/linkedin-brand.svg";
import { ReactComponent as GithubLogo } from "../../assets/github-brand.svg";
import { Text, ContainerWrapper } from "../../styles";

const Navbar = () => {
  return (
    <StyledWrapper>
      <Wrapper>
        <OzielLogo />
        <LogoText style={{ marginLeft: 24 }}>pageinspector</LogoText>
      </Wrapper>
      <Wrapper /* style={{marginTop: 5}} */>
        <a
          style={{ textDecoration: "none" }}
          href="https://www.linkedin.com/in/ozielalves/"
          title="Linkedin - Oziel Alves"
        >
          <LinkedinIcon />
        </a>
        <a
          style={{ textDecoration: "none", marginLeft: 24 }}
          href="https://github.com/ozielalves"
          title="GitHub - Oziel Alves"
        >
          <GithubIcon style={{ marginLeft: 24 }} />
        </a>
      </Wrapper>
    </StyledWrapper>
  );
};

const StyledWrapper = styled(ContainerWrapper)`
  height: 73px;
  justify-content: space-between;
  padding: 0 48px;
  border: 1px solid var(--base-color-border);
  border-left: none;
  border-right: none;
  border-top: none;
`;

const Wrapper = styled(ContainerWrapper)`
  width: auto;
  height: auto;
  display: flex;
  justify-content: center;
  align-items: center;
`;

const LogoText = styled(Text)`
  font-weight: bold;
  font-size: 26.6967px;
  line-height: 33px;

  @media (max-width: 600px) {
    font-size: 1.2rem;
  }
`;

const LinkedinIcon = styled(LinkedinLogo)`
  transition: opacity 0.3s ease 0s;
  &:hover {
    opacity: 0.7;
  }
`;

const GithubIcon = styled(GithubLogo)`
  transition: opacity 0.3s ease 0s;
  &:hover {
    opacity: 0.7;
  }
`;

export default Navbar;
