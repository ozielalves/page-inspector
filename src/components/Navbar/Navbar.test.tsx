import renderer from "react-test-renderer";
import React from "react";
import Navbar from "./index";

describe("Navbar render test", () => {
  it("should render the correct Navbar correctly", () => {
    const tree = renderer
      .create(<Navbar/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
