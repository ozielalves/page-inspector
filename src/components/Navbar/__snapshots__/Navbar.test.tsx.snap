// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar render test should render the correct Navbar correctly 1`] = `
<div
  className="sc-jSgupP sc-iqHYGH gmVoiw iZHmQk"
>
  <div
    className="sc-gKsewC eygLjf"
    style={
      Object {
        "justifyContent": "flex-start",
      }
    }
  >
    <svg>
      logo-navbar-oziel.svg
    </svg>
    <p
      className="sc-gsTCUz sc-dlfnbm hFbaQT kWkuPj"
      style={
        Object {
          "marginLeft": 24,
        }
      }
    >
      pageinspector
    </p>
  </div>
  <div
    className="sc-gKsewC sc-kEjbxe eygLjf eJdoup"
  >
    <a
      href="https://www.linkedin.com/in/ozielalves/"
      style={
        Object {
          "margin": "5px 0 0 24px",
          "textDecoration": "none",
        }
      }
      title="Linkedin - Oziel Alves"
    >
      <svg
        className="sc-crrsfI dHhAbk"
      >
        linkedin-brand.svg
      </svg>
    </a>
    <a
      href="https://github.com/ozielalves"
      style={
        Object {
          "margin": "5px 0 0 24px",
          "textDecoration": "none",
        }
      }
      title="GitHub - Oziel Alves"
    >
      <svg
        className="sc-dQppl etXCAW"
      >
        github-brand.svg
      </svg>
    </a>
  </div>
</div>
`;
